WordPress/gutenberg
# Responsive Navigation

## Description

This PR introduces a mobile responsive Navigation Block. It add the markup necessary to display the Navigation Menu as a floating modal when the resolution of the site is below our mobile breakpoint.

Functionality in this PR depends on #29537 in order to add a library which enables toggling the visibility of the Menu, as well as creating an accessible experience. Here I am working under the assumption that this will be a possibility in the near future.

### Next steps, and known issues

- [x] Get accessibility working — it did not work out of the box. No guesses yet as to why this was the case.
- [x] Proper UI.
- [x] Toggling responsiveness on and off.
- [x] Wrap the Navigation container in the modal markup only if responsiveness has been turned on.
- [x] Custom styling — I haven't thought that far ahead at this point, but I'm assuming that at the very least we'd like to support the current overlay version, and a slide down menu.

### Testing steps
- Checkout this branch.
- Create a navigation menu.
- Make your browser screen smaller, below `480px`.
- A button labeled "Open Menu" should appear.
- After clicking that button, elements should be shown vertically on smaller resolutions
- A button should appear in the top right corner of the screen, which allows closing the modal/overlay.

### Small demo

https://user-images.githubusercontent.com/1157901/111841533-dacd4380-88dc-11eb-962d-b7aec823686c.mov
